### PR TITLE
feat(utxo-bin): add new command convertTx

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
- feat(blockapis): add fetchPrevTxBuffers
- feat(utxo-bin): use strict mode
- feat(utxo-bin): add new command convertTx

### CODE REVIEW

File: modules/utxo-bin/src/commands.ts [hunk 1]
```diff
@@ -32,16 +32,115 @@ import {
...
+    if (data) {
+      throw new Error(`conflicting arguments`);
+    }
+    data = argv.data;
+  }
+
+  // strip whitespace
+  if (!data) {
+    throw new Error(`no txdata`);
+  }
```
Comment (rushilbg) [inline modules/utxo-bin/src/commands.ts:115]: Possible to move all this in a separate function `validateArguments` or such? This whole function will become easier to reason about then, it's quite long currently.

File: modules/utxo-bin/src/convertTransaction.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+): Promise<void> {
+  const { format } = params;
+
+  switch (format) {
+    case 'legacy':
+      if (tx instanceof utxolib.bitgo.UtxoTransaction) {
+        throw new Error(`input is already in legacy format`);
+      }
+      if (tx instanceof utxolib.bitgo.UtxoPsbt) {
+        throw new Error(`TODO`);
```
Comment (rushilbg) [inline modules/utxo-bin/src/convertTransaction.ts:42]:
```
throw new Error(`not implemented yet`);

File: modules/utxo-bin/src/convertTransaction.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+      if (tx instanceof utxolib.bitgo.UtxoTransaction) {
+        const psbt = await legacyToPsbt(params.httpClient, tx);
+        const { outfile = getOutfile(tx, format) } = params;
+        await fs.writeFile(outfile, psbt.toBase64());
+        console.log(`wrote ${outfile}`);
+        return;
+      }
+
+      break;
```
Comment (rushilbg) [inline modules/utxo-bin/src/convertTransaction.ts:58]: Redundant?

Comment (OttoAllmendinger) [reply modules/utxo-bin/src/convertTransaction.ts]: improved this case

File: modules/utxo-bin/src/commands.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/utxo-bin/src/commands.ts:115]: split it up into two methods

File: modules/utxo-bin/src/convertTransaction.ts (hunk 2 above)
Comment (OttoAllmendinger) [reply modules/utxo-bin/src/convertTransaction.ts]: done

**Review by rushilbg — APPROVED** [review]

**Pull request closed without merging**